contracts/Governance.sol

```diff
@@ -0,0 +1,285 @@
+    }
+
+
+    /** @dev Put proposal to support voting only when a new proposal is permitted.
+     *  @param _id ID of a proposal
+     */
+    function putProposalToSupport(bytes32 _id) public onlyWhenProposalIsNew(_id) onlyWhenPermitted(_id) {
+        Proposal storage proposal = proposals[_id];
+
+        address cloneToken = pinakion.createCloneToken({_cloneTokenName: quorumTokenName, _cloneDecimalUnits: DECIMALS, _cloneTokenSymbol: quorumTokenSymbol, _snapshotBlock: block.number, _transfersEnabled: true});
+        proposal.quorumToken = MiniMeToken(cloneToken);
+        proposal.quorumToken.changeController(tokenController);
+
+        proposal.state = ProposalState.PutToSupport;
+
+        emit ProposalPutToSupport(_id);
+    }
+
+
+    /** @dev Calculate and return required quorum for a given proposal.
+     *  @param _id ID of a proposal
+     */
+    function getRequiredQuorum(bytes32 _id) public view onlyWhenProposalPutToSupport(_id) returns (uint effectiveQuorum){
+        uint numberOfDividePeriodsPassed = (block.timestamp - lastTimeQuorumReached) / quorumDivideTime;
+        effectiveQuorum = currentProposalQuorum * proposals[_id].quorumToken.totalSupply() / (2 ** numberOfDividePeriodsPassed) / 100;
+    }
+
+
+    /** @dev Put given proposal to vote.
+     *  @param _id ID of a proposal
+     */
+    function putProposalToVote(bytes32 _id) public onlyWhenProposalPutToSupport(_id) onlyWhenQuorumReached(_id) {
+        Proposal storage proposal = proposals[_id];
+
+        proposal.whenPutToVote = block.timestamp;
+
+        address cloneToken = pinakion.createCloneToken({_cloneTokenName: voteTokenName, _cloneDecimalUnits: DECIMALS, _cloneTokenSymbol: voteTokenSymbol, _snapshotBlock: block.number, _transfersEnabled: true});
+        proposal.voteToken = MiniMeToken(cloneToken);
+        proposal.voteToken.changeController(tokenController);
+
+        proposal.state = ProposalState.PutToVote;
+
+        emit ProposalPutToVote(_id);
+
+        resetSettings(); // Reset global variables as quorum has reached
+    }
+
+
+    /** @dev Ends a voting, moves proposal to decided state, sets the decision
+     *  @param _id ID of a proposal
+     */
+    function finalizeVoting(bytes32 _id) public onlyWhenProposalPutToVote(_id) onlyWhenVotingPeriodPassed(_id) {
+        proposals[_id].state = ProposalState.Decided;
+        proposals[_id].approved = proposals[_id].voteToken.balanceOf(approvalDeposit) > proposals[_id].voteToken.balanceOf(rejectionDeposit);
+
+        emit ProposalDecided(_id, proposals[_id].approved);
+    }
+
+
+    /** @dev General purpose call function for executing a proposal UNTRUSTED
+     *  @param _id ID of a proposal
+     */
+    function executeProposal(bytes32 _id) public onlyWhenApproved(_id) {
+        Proposal storage proposal = proposals[_id];
+
+        proposal.destination.call.value(proposal.amount)(proposal.data); // solium-disable-line security/no-call-value
```

Comment:
Where was the value for `proposal.amount` provided? We need to require it somewhere along the flow.

Reply:
Or will we top off the contract's balance every so often?

Reply:
Three options come to my mind:

1. Make proposer pay when creating a proposal.
2. Let the contract pay as you said.
3. Collect the required amount from voters who approved. This is complicated much more complicated than the first and the second option.

What do you think? @satello @n1c01a5 @clesaege

Reply:
I think 2 makes more sense. One party should not have to fund the organization's decisions.